plumbing/format/idxfile/idxfile.go

```diff
@@ -213,22 +204,19 @@ func (idx *MemoryIndex) genOffsetHash() error {
 	idx.offsetHash = make(map[int64]plumbing.Hash, count)
 	idx.offsetHashIsFull = true
 
-	iter, err := idx.Entries()
-	if err != nil {
-		return err
-	}
-
-	for {
-		entry, err := iter.Next()
-		if err != nil {
-			if err == io.EOF {
-				return nil
-			}
-			return err
+	var hash plumbing.Hash
+	i := uint32(0)
+	for firstLevel, fanoutValue := range idx.Fanout {
+		mappedFirstLevel := idx.FanoutMapping[firstLevel]
+		for secondLevel := uint32(0); i < fanoutValue; i++ {
+			copy(hash[:], idx.Names[mappedFirstLevel][secondLevel*objectIDLength:])
+			offset := int64(idx.getOffset(mappedFirstLevel, int(secondLevel)))
+			idx.offsetHash[offset] = hash
+			secondLevel++
```

Comment:
Is this a leftover?

Reply:
Nope. The loop is a bit weird since it's iterating over two variables. Maybe there's some better Go syntax for it?

`secondLevel` goes from 0 to maximum value on that fanout level (not checked directly). `i` goes from 0 over both the inner and outer loop and bounds the inner loop.